In [41]:
from tensorflow.keras.preprocessing.text import one_hot

In [42]:
### sentences
sent=[  'the glass of milk',
     'the glass of juice',
     'the cup of tea',
    'I am a good boy',
     'I am a good developer',
     'understand the meaning of words',
     'your videos are good',]

In [43]:
sent

['the glass of milk',
 'the glass of juice',
 'the cup of tea',
 'I am a good boy',
 'I am a good developer',
 'understand the meaning of words',
 'your videos are good']

In [44]:
## Define the vocabulary size
voc_size=10000

In [45]:
one_hot_repr = [one_hot(word, voc_size) for word in sent]
one_hot_repr

[[9920, 3261, 9790, 6385],
 [9920, 3261, 9790, 4385],
 [9920, 557, 9790, 4943],
 [6658, 6532, 5528, 1940, 8542],
 [6658, 6532, 5528, 1940, 384],
 [2657, 9920, 1024, 9790, 8818],
 [5150, 6389, 5593, 1940]]

y index dekhre hain 9920 means is index pr 1 baaki zero agar tu pehli aur dusri line compare kre to dikhajyagea tuje pehle 3 to same hi hain

In [46]:
sent

['the glass of milk',
 'the glass of juice',
 'the cup of tea',
 'I am a good boy',
 'I am a good developer',
 'understand the meaning of words',
 'your videos are good']

In [47]:
## word Embedding Representation
from tensorflow.keras.layers import Embedding
#from tensorflow.keras.processing.sequence import pad_sequences
from tensorflow.keras.utils import pad_sequences
from tensorflow.keras.models import Sequential

In [48]:
import numpy as np

In [ ]:
sent_len = 8
embedded_docs = pad_sequences(one_hot_repr,padding='pre',maxlen=sent_len)

equal length mein krna pada kyoki hr sentence ki length diff thi max 5 hain sentence ki lenhgth to uske acc maine 8 select krke sbko 8 k brabar krdiya

In [50]:
embedded_docs

array([[   0,    0,    0,    0, 9920, 3261, 9790, 6385],
       [   0,    0,    0,    0, 9920, 3261, 9790, 4385],
       [   0,    0,    0,    0, 9920,  557, 9790, 4943],
       [   0,    0,    0, 6658, 6532, 5528, 1940, 8542],
       [   0,    0,    0, 6658, 6532, 5528, 1940,  384],
       [   0,    0,    0, 2657, 9920, 1024, 9790, 8818],
       [   0,    0,    0,    0, 5150, 6389, 5593, 1940]], dtype=int32)

In [51]:
embedded_docs.shape



(7, 8)

In [52]:
dim=10

In [53]:
model = Sequential()
model.add(Embedding(voc_size,dim,input_length=sent_len))
model.compile('adam','mse')

In [54]:
model.summary()

Model: "sequential_3"

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┓
┃ Layer (type)                    ┃ Output Shape           ┃       Param # ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━┩
│ embedding_3 (Embedding)         │ ?                      │   0 (unbuilt) │
└─────────────────────────────────┴────────────────────────┴───────────────┘

 Total params: 0 (0.00 B)

 Trainable params: 0 (0.00 B)

 Non-trainable params: 0 (0.00 B)

In [55]:
model.predict(embedded_docs)

1/1 ━━━━━━━━━━━━━━━━━━━━ 0s 46ms/step


array([[[ 1.41532682e-02, -4.33609374e-02,  4.81327623e-03,
         -1.65938251e-02,  4.61277105e-02,  3.89008559e-02,
         -2.88696773e-02, -4.92256284e-02,  1.50025524e-02,
         -2.96467897e-02],
        [ 1.41532682e-02, -4.33609374e-02,  4.81327623e-03,
         -1.65938251e-02,  4.61277105e-02,  3.89008559e-02,
         -2.88696773e-02, -4.92256284e-02,  1.50025524e-02,
         -2.96467897e-02],
        [ 1.41532682e-02, -4.33609374e-02,  4.81327623e-03,
         -1.65938251e-02,  4.61277105e-02,  3.89008559e-02,
         -2.88696773e-02, -4.92256284e-02,  1.50025524e-02,
         -2.96467897e-02],
        [ 1.41532682e-02, -4.33609374e-02,  4.81327623e-03,
         -1.65938251e-02,  4.61277105e-02,  3.89008559e-02,
         -2.88696773e-02, -4.92256284e-02,  1.50025524e-02,
         -2.96467897e-02],
        [ 8.92424583e-03, -1.66152604e-02,  3.46038081e-02,
          3.36939804e-02,  2.47050449e-03, -2.03207619e-02,
         -2.53042337e-02, -9.53428820e-03, -4.004237

In [56]:
embedded_docs[0]

array([   0,    0,    0,    0, 9920, 3261, 9790, 6385], dtype=int32)

In [57]:
model.predict(embedded_docs[0:1])

1/1 ━━━━━━━━━━━━━━━━━━━━ 0s 14ms/step


array([[[ 0.01415327, -0.04336094,  0.00481328, -0.01659383,
          0.04612771,  0.03890086, -0.02886968, -0.04922563,
          0.01500255, -0.02964679],
        [ 0.01415327, -0.04336094,  0.00481328, -0.01659383,
          0.04612771,  0.03890086, -0.02886968, -0.04922563,
          0.01500255, -0.02964679],
        [ 0.01415327, -0.04336094,  0.00481328, -0.01659383,
          0.04612771,  0.03890086, -0.02886968, -0.04922563,
          0.01500255, -0.02964679],
        [ 0.01415327, -0.04336094,  0.00481328, -0.01659383,
          0.04612771,  0.03890086, -0.02886968, -0.04922563,
          0.01500255, -0.02964679],
        [ 0.00892425, -0.01661526,  0.03460381,  0.03369398,
          0.0024705 , -0.02032076, -0.02530423, -0.00953429,
         -0.04004238,  0.04595227],
        [ 0.00526174,  0.04890246, -0.03981184,  0.04263024,
         -0.0065494 , -0.03196293,  0.01842341, -0.03930848,
         -0.03733633,  0.01369261],
        [ 0.03315735, -0.03795435,  0.032816  , -0.0